posts/deploying-a-jhipster-app-to-aws-using-elastic-beanstalk.md

```diff
@@ -0,0 +1,101 @@
+
+In this blog, I will focus on AWS and especially on Elastic Beanstalk which is an orchestration service. Elastic Beanstalk helps you create a production-ready infrastructure that has the architecture below:
+
+![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/diagram-eb.png)
+
+# App generation with JHipster
+
+Currently the Elastic Beanstalk sub-generator only supports SQL databases (Oracle and Microsoft SQL Server excluded). Our JHipster app will be a basic monolith that uses React and PostgreSQL as a database.
+
+```json
+{
+  "generator-jhipster": {
+    "promptValues": {
+      "packageName": "com.company",
+      "nativeLanguage": "en"
+    },
+    "jhipsterVersion": "6.4.1",
+    "applicationType": "monolith",
+    "baseName": "jhipster",
+    "packageName": "com.company",
+    "packageFolder": "com/company",
+    "serverPort": "8080",
+    "authenticationType": "jwt",
+    "cacheProvider": "ehcache",
+    "enableHibernateCache": true,
+    "websocket": false,
+    "databaseType": "sql",
+    "devDatabaseType": "h2Disk",
+    "prodDatabaseType": "postgresql",
+    "searchEngine": false,
+    "messageBroker": false,
+    "serviceDiscoveryType": false,
+    "buildTool": "maven",
+    "enableSwaggerCodegen": false,
+    "embeddableLaunchScript": false,
+    "useSass": true,
+    "clientPackageManager": "npm",
+    "clientFramework": "react",
+    "clientTheme": "none",
+    "clientThemeVariant": "",
+    "testFrameworks": [],
+    "jhiPrefix": "jhi",
+    "entitySuffix": "",
+    "dtoSuffix": "DTO",
+    "otherModules": [],
+    "enableTranslation": true,
+    "nativeLanguage": "en",
+    "languages": [
+      "en"
+    ],
+    "blueprints": []
+  }
+}
+```
+
+The `.yo-rc.json` above can be used to avoid answering all the questions manually.
+
+# Deployment to AWS with Elastic Beanstalk
+
+ First, make sure that your AWS CLI is [correctly configured](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html#cli-quick-configuration) and the account must have the `AWSElasticBeanstalkFullAccess` policy. If you use a named profile different than `default`, you just have to set the environment variable `AWS_PROFILE` with the correct profile.
+
+ The command `jhipster aws` will ask questions about the AWS deployment:
+
+ ![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/jhipster-aws.png)
+
+ The AWS account used needs access to the S3 bucket since it will be used as a storage location for the application's Jars. The sub-generator will also create a RDS instance so make sure that the database credentials are correct and match the ones in `application-prod.yml`. The application will always be packaged using the `prod` profile, which means that all the tests must be green!
```

Comment:
```suggestion
 The AWS account used needs access to the S3 bucket since it will be used as a storage location for the application's Jars. The sub-generator will also create a RDS instance so make sure that the database credentials are correct and match the ones in `application-prod.yml` located in the root of your JHipster app. The application will always be packaged using the `prod` profile, which means that all the tests must be green!
```
Also better explain the connection to Elastic Beanstalk with the packaging using the `prod` profile and the tests that must be green.